caikit/runtime/service_generation/data_stream_source.py

```diff
@@ -450,6 +451,13 @@ def name_to_plugin_map(self):
             plugin.get_field_name(self.ELEMENT_TYPE): plugin for plugin in self.PLUGINS
         }
 
+    @cached_property
+    def _stream(self):
```

Comment:
Do we need to worry about OOMs here for large data sets?

Reply:
Well, if `_to_data_stream` is loading a bunch of data into memory then it'd still be better to only do it once, right? 

I think it'd be up to the plugin authors to make sure that large datasets are properly handled. The WIP plugin we have now is loading datasets into memory after downloading, but we would probably write into a tempfile instead if we ever configured it to allow large downloads

Reply:
Make sense!